app/models/rubygem.rb

```diff
@@ -188,6 +188,13 @@ def owned_by?(user)
     ownerships.exists?(user_id: user.id)
   end
 
+  def owned_by_with_role?(user, minimum_required_role)
+    return false if user.blank?
+    ownerships.exists?(["user_id = ? AND role >= ?", user.id, Access.flag_for_role(minimum_required_role)])
```

Comment:
let's try to avoid writing raw SQL outside of the model being queried.

this could be something like `ownerships.where(user:).minimum_role(minimum_required_role).exists?`

and then ownership could have a `scope :minimum_role, -> (role) { where("role >= ?", ...`

Reply:
Good callout. I've added a scope and replaced the raw sql.